test/authentication-test.ts

```diff
@@ -0,0 +1,120 @@
+import axios from "axios";
+import bcrypt from "bcrypt";
+import { expect } from "chai";
+import jwt from "jsonwebtoken";
+import { prisma } from "../src/client/client";
+
+const port = process.env.PORT ? +process.env.PORT : 30002;
+const JWT_SECRET = process.env.JWT_SECRET!;
+
+
+describe("POST /auth", () => {
+  it("Should return a valid token and user info on successful login", async () => {
+    const mockUser = {
+      name: "testuser",
+      email: "testuser@example.com",
+      password: "Password123",
+      birthDate: new Date("2000-01-01"),
+    };
+
+    const hashedPassword = await bcrypt.hash(mockUser.password, 10);
+
+    await prisma.user.create({
+      data: { ...mockUser, password: hashedPassword },
+    });
+
+    const response = await axios.post(`http://localhost:${port}/auth`, {
+      email: mockUser.email,
+      password: mockUser.password,
+    });
+
+    expect(response.status).to.equal(201);
+
+    const { user, token } = response.data;
+
+    expect(user).to.deep.equal({
+      id: user.id,
+      name: mockUser.name,
+      email: mockUser.email,
+      birthDate: mockUser.birthDate.toISOString(),
+    });
+
+    const decodedToken: any = jwt.verify(token, JWT_SECRET);
+
+    expect(decodedToken.id).to.be.deep.eq(user.id);
+    const now = Math.floor(Date.now() / 1000);
+    expect(decodedToken.exp - now).to.be.closeTo(24 * 3600, 5);
+  });
+
+  it("Should return a valid token and expiration time of 1 day", async () => {
```

Comment:
```suggestion
  it("Should return a valid token and of long expiration", async () => {
```

A sugestão é só pra tirar o "1 dia" do enunciado, porque pode ser que esse tempo varie.